src/tools/schema2kotlin.ts

```diff
@@ -184,6 +184,8 @@ ${imports.join('\n')}
       } else if (type.isTuple) {
         const innerTypes = type.getContainedTypes();
         return `Tuple${innerTypes.length}<${innerTypes.map(t => generateInnerType(t)).join(', ')}>`;
+      } else if (type.isVariable) {
+        return generateInnerType(type.canWriteSuperset);
```

Comment:
I presume that whether `canWriteSuperset` or `canReadSubset` is filled in depends on the direction of the connection (read vs write), so maybe you should do something like: `type.canWriteSuperset || type.canReadSubset` ?

Also what happens if there are no restrictions, i.e.

```
particle P
  x: reads ~a
```

This should work and the code generated entity should have 0 accessible fields. In other words it is equivalent to: `~a with {}`, so probably you need something like: type.canWriteSuperset || type.canReadSubset || EMPTY_SCHEMA`, I am not sure whether we have a constant for an empty schema, you can create one.

Reply:
Saving generation for a future PR.